File: task-manager-front/src/api/interfaces/userService.ts
import { userHttp } from "../http";

export interface UserBoundary {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    phoneNumber: string;

}
export type UserSearchResult = {
  id: string;
  firstName: string;
  lastName: string;
};
export const userService = {
    createUser: async (user: UserBoundary) => {
        const response = await userHttp.post("/create", user);
        return response.data;
    },
    getUserById: async (userId: string) => {
        const response = await userHttp.get(`/get/${userId}`);
        return response.data;
    },
    updateUser: async (userId: string, updatedUser: UserBoundary) => {
        const response = await userHttp.put(`/update/${userId}`, updatedUser);
        return response.data;
    },
    deleteUser: async (userId: string) => {
        const response = await userHttp.delete(`/delete/${userId}`);
        return response.data;
    },
    searchUsers: async (query: string) => {
        const response = await userHttp.get(`/search?query=${query}`);
        return response.data;
    }
    
    
};

File: task-manager/src/main/java/com/server/task_manager/userModule/userService/UserService.java
package com.server.task_manager.userModule.userService;

import java.util.List;

import com.server.task_manager.userModule.userBoundary.UserBoundary;
import com.server.task_manager.userModule.userEntity.UserEntity;

public interface UserService {
    UserEntity createUser(UserBoundary userBoundary);
    UserEntity getUserById(String userId);
    UserEntity updateUser(String userId, UserBoundary userBoundary);
    void deleteUser(String userId);
    List<UserEntity> getAllUsers();
    void deleteAllUsers();
    List<UserEntity> searchUsers(String query);
}
